REST/PowerShell/Variables/AddOrEditVariablesWithScoping.ps1

```diff
@@ -142,6 +142,136 @@ Function Add-Variable {
+        'Value' = $($VariableValue)
+        'Type' = 'String'
+        'IsSensitive' = $IsSensitive
+        'Scope' = @{
+          'Environment' =@()
+          'Role' =@()
+        }
+    }
+
+    if ($VariableRoleScope){
+        #If the role passed was an array, add them all
+        if ($VariableRoleScope -is [array]){
+            foreach ($role in $VariableRoleScope){
+                $obj.scope.role += $VariableRoleScope
+            }
+        }
+        #If it's not an array, just add the one.
+        else{
+            $obj.scope.role += $VariableRoleScope
+        }
+    }
+
+    if ($VariableEnvScope){
+        #If the environment passed was an array, add them all
+        if ($VariableEnvScope -is [array]){
+            foreach ($environment in $variableenvscope){
+                $environmentObj = $VariableSet.ScopeValues.Environments | Where { $_.Name -eq $environment } | Select -First 1
+                $obj.scope.Environment += $environmentObj.Id
+            }
+        }
+        #If it's not an array, just add the one.
+        else{
+            $environmentObj = $VariableSet.ScopeValues.Environments | Where { $_.Name -eq $VariableEnvScope } | Select -First 1
+            $obj.scope.environment += $environmentObj.Id
+        }
+    }
+    #iterate each variable to match on the one we want to modify
+    foreach ($variable in $VariableSet.Variables){
+        $envmatch = $null
+        $rolematch = $null
+        $temprolelist = @()
+        $tempenvlist = @()
+        $tempExistList = @()
+        $envMatchBool = $null
+        $roleMatchBool = $null
+        #create list of environments based on IDs
+        foreach($env in $ExistingEnvScope){
+            $tempId = $VariableSet.ScopeValues.Environments | Where { $_.Name -eq $env } | Select -First 1
+            $tempExistList += $tempId.Id
+        }
+
+        #put the scopes in a format we can compare
+        foreach ($env in $variable.Scope.Environment){
+            $tempenvlist += $env
+        }
+        #sort to compare
+        $tempenvlist = $tempenvlist | sort
+        $tempExistList = $tempExistList | sort
+        #test compare
+        if (!$null -eq $tempenvlist -and !$null -eq $tempExistList){
+        $envMatchBool = Compare-Object -ReferenceObject $tempenvlist -DifferenceObject $tempExistList -PassThru
+        }
+        if ($null -eq $envMatchBool){
+            $envmatch = $true
+        }
+        else{
+            $envmatch = $false
+        }
+        #same as above but for roles
+        foreach ($role in $variable.Scope.Role){
+            $temprolelist += $role
+        }
+        $temprolelist = $temprolelist | sort
+        $ExistingRoleScope = $ExistingRoleScope | sort
+
+        if (!$null -eq $temprolelist -and !$null -eq $ExistingRoleScope){
+            $roleMatchBool = Compare-Object -ReferenceObject $temprolelist -DifferenceObject $ExistingRoleScope -PassThru
+            }
+
+        if ($null -eq $roleMatchBool){
```

Comment:
Same point here for the `$rolematchBool`. If its null, doesn't that mean there isn't a match? It can be simplified to one line regardless of my understanding :smile:

Reply:
Good call on changing it. Changed all instances of $roleMatchBool to $roleDiffs.